heka/sandbox/filters/telemetry_decoder_view.lua

```diff
@@ -79,19 +80,17 @@ function process_message ()
                 return -1, "Decoder is missing its name"
             end
 
-            local id = string.match(v.Name, "^TelemetryKafkaInput(%d+)")
-            if not id then
-                return -1, "Telemetry decoder is missing its identifier"
-            else
+            local id = string.match(v.Name, decoder_match)
+            if id then
                 id = tonumber(id)
-            end
 
-            if type(v["ProcessMessageCount-TelemetryDecoder"]) == "table" then
-                update_delta(ts, TOTAL, id, id_count, v["ProcessMessageCount-TelemetryDecoder"].value)
-            end
+                if type(v["ProcessMessageCount-TelemetryDecoder"]) == "table" then
+                    update_delta(ts, TOTAL, id, id_count, v["ProcessMessageCount-TelemetryDecoder"].value)
+                end
 
-            if type(v["ProcessMessageFailures-TelemetryDecoder"]) == "table" then
-                update_delta(ts, FAILURES, id, id_failures, v["ProcessMessageFailures-TelemetryDecoder"].value)
+                if type(v["ProcessMessageFailures-TelemetryDecoder"]) == "table" then
+                    update_delta(ts, FAILURES, id, id_failures, v["ProcessMessageFailures-TelemetryDecoder"].value)
+                end
             end
```

Comment:
Yes, since there is only one match string, Previously the error message "id" is missing was misleading in some cases because the other decoder name didn't match the pattern at all. We could make it more explicit with two matches (one for the prefix and one for the id), but I don't think it's worth it.